pkg/controller/replication/replication_controller.go

```diff
@@ -580,17 +639,48 @@ func (rm *ReplicationManager) syncReplicationController(key string) error {
 	trace.Step("ReplicationController restored")
 	rcNeedsSync := rm.expectations.SatisfiedExpectations(rcKey)
 	trace.Step("Expectations restored")
-	podList, err := rm.podStore.Pods(rc.Namespace).List(labels.Set(rc.Spec.Selector).AsSelector())
 	if err != nil {
 		glog.Errorf("Error getting pods for rc %q: %v", key, err)
 		rm.queue.Add(key)
 		return err
 	}
 	trace.Step("Pods listed")
 
-	// TODO: Do this in a single pass, or use an index.
-	filteredPods := controller.FilterActivePods(podList.Items)
-	if rcNeedsSync {
+	// TODO: Do the List and Filter in a single pass, or use an index.
+	var filteredPods []*api.Pod
+	if rm.garbageCollectorEnabled {
+		// list all pods to include the pods that don't match the rc's selector
+		// anymore but has the stale controller ref.
+		podList, err := rm.podStore.Pods(rc.Namespace).List(labels.Everything())
+		if err != nil {
+			glog.Errorf("Error getting pods for rc %q: %v", key, err)
+			rm.queue.Add(key)
+			return err
+		}
+		var abandonPods []*api.Pod
+		filteredPods, abandonPods = controller.ControlActivePods(
```

Comment:
nit: can you just construct a custom type here? 
```
foo := NewOwnerRefManagerForRC(rm.podControl, rc)
filtered, abandoned := foo.ControlActivePods(podList.Items)
```

I don't have a huge preference, but if we want to grow the surface area of controllerRef having a well defined set of public functions that controllers can call, and an obvious class to add new methods into, makes maintaining it as a library easier.